src/clincoded/static/components/add_external_resource.js

```diff
@@ -735,32 +751,42 @@ function carQueryResource() {
         var url = this.props.protocol + external_url_map['CARallele'];
         var data;
         var id = this.state.inputValue;
-        this.getRestData(url + id).then(json => {
+        this.getRestData(url + id).then(async (json) => {
+            // final state values after query; default to nothing happened
+            let finalState = {
+                queryResourceBusy: false,
+                resourceFetched: true
+            }
+
             data = parseCAR(json);
+            finalState.tempResource = data;
+            
             if (data.clinvarVariantId) {
                 // if the CAR result has a ClinVar variant ID, query ClinVar with it, and use its data
                 url = external_url_map['ClinVarEutilsVCV'];
-                this.getRestDataXml(url + data.clinvarVariantId).then(xml => {
+                try {
+                    const xml = await this.getRestDataXml(url + data.clinvarVariantId);
                     var data_cv = parseClinvar(xml);
                     if (data_cv.clinvarVariantId) {
                         // found the result we want
                         data_cv.carId = id;
-                        this.setState({queryResourceBusy: false, tempResource: data_cv, resourceFetched: true});
+                        finalState.tempResource = data_cv;
                     } else {
-                        // something failed with the parsing of ClinVar data; roll back to CAR data
-                        this.setState({queryResourceBusy: false, tempResource: data, resourceFetched: true});
+                        // something failed with the parsing of ClinVar data; roll back to CAR data (just use initial value of `finalState`)
                     }
-                }).catch(e => {
+                }
+                catch (e) {
                     // error handling for ClinVar query
                     this.setFormErrors('resourceId', 'Error querying ClinVar for additional data. Please check your input and try again.');
-                    this.setState({queryResourceBusy: false, resourceFetched: false});
-                });
+                    finalState.resourceFetched = false;
```

Comment:
To maintain previous `setState` logic, I'd recommend adding a removal of `tempResource` (from `finalState`) here.